components/app.js

```diff
@@ -0,0 +1,129 @@
+function getAverage(){
+  var total = 0
+  for(var i = 0; i < gradeData.length; i++){
+    total = total += parseInt(gradeData[i].grade, 10)
+  }
+  var avg = total / gradeData.length
+  return avg.toFixed(2)
+}
+
+class App {
+  handleGetGradesError(error) {
+    console.error(error)
+  }
+  handleGetGradesSuccess(grades){
+    gradeData = grades
+    this.gradeTable.updateGrades()
+    var gradeAverage = getAverage()
+    this.pageHeader.updateAverage(gradeAverage)
+  }
+  constructor(gradeTable, pageHeader, gradeForm, updateGradeForm) {
+    this.gradeTable = gradeTable
+    this.handleGetGradesSuccess = this.handleGetGradesSuccess.bind(this)
+    this.handleGetGradesError = this.handleGetGradesError.bind(this)
+    this.createGrade = this.createGrade.bind(this)
+    this.handleCreateGradeError = this.handleCreateGradeError.bind(this)
+    this.handleCreateGradeSuccess = this.handleCreateGradeSuccess.bind(this)
+    this.deleteGrade = this.deleteGrade.bind(this)
+    this.handleDeleteGradeError = this.handleDeleteGradeError.bind(this)
+    this.deleteGradeSuccess = this.deleteGradeSuccess.bind(this)
+    this.pageHeader = pageHeader
+    this.addGradeForm = addGradeForm
+    this.updateGradeForm = updateGradeForm
+    this.handleUpdateGradeError = this.handleUpdateGradeError.bind(this)
+    this.handleUpdateGradeSuccess = this.handleUpdateGradeSuccess.bind(this)
+    this.updateGrade = this.updateGrade.bind(this)
+  }
+  getGrades(){
+    $.ajax({
+      headers: {
+        "X-Access-Token": "NUEI5rKm"
+      },
+      type: "GET",
+      url: "https://sgt.lfzprototypes.com/api/grades",
+      error: this.handleGetGradesError,
+      success: this.handleGetGradesSuccess,
+    })
+  }
+  start(){
+    this.getGrades()
+    this.addGradeForm.onSubmit(this.createGrade)
+    this.updateGradeForm.onSubmitUpdate(this.updateGrade)
+    this.gradeTable.onDeleteClick(this.deleteGrade)
+  }
+  createGrade(name, course, grade) {
+    $.ajax({
+      headers: {
+        "X-Access-Token": "NUEI5rKm"
+      },
+      type: "POST",
+      url: "https://sgt.lfzprototypes.com/api/grades",
+      data: {
+        name: name,
+        course: course,
+        grade: grade
+      },
+      success: this.handleCreateGradeSuccess,
+      error: this.handleCreateGradeError
+    })
+  }
+  handleCreateGradeError(error) {
+    console.error(error)
+  }
+  handleCreateGradeSuccess(grades){
+    gradeData.push(grades)
+    var gradeAverage = getAverage()
+    this.pageHeader.updateAverage(gradeAverage)
+    this.gradeTable.updateGrades()
+  }
+  deleteGradeSuccess(){
+    var gradeAverage = getAverage()
+    this.pageHeader.updateAverage(gradeAverage)
+    this.gradeTable.updateGrades()
+  }
+  deleteGrade(id) {
+    $.ajax({
+      headers: {
+        "X-Access-Token": "NUEI5rKm"
+      },
+      type: "DELETE",
+      url: "https://sgt.lfzprototypes.com/api/grades/" + id,
+      success: this.deleteGradeSuccess,
+      error: this.handleDeleteGradeError
+    })
+  }
```

Comment:
Is the grade still being removed from the page after the API responds successfully?

What you had before was close: the `deleteGradeSuccess` method still needs the `id` of the grade that's been removed, so the `deleteGrade` method still needs to specify the `id`, What you want to do is re-`bind` the `deleteGradeSuccess` method inside `deleteGrade`, but pass `bind` an additional argument to be passed to the `deleteGradeSuccess` method when it is eventually called:
```js
success: this.deleteGradeSuccess.bind(undefined, id),
```
_Note: since this method was already bound in the_ `constructor`_, the value of `this` won't change if you_ `bind` _it again, hence the_ `undefined` _first argument._

Reply:
The grade is being removed from the page. The id is still being send with the DELETE request so it's taken off the grades database and since my table is now built off information stored in an array I just changed the function to delete the data from the array when the delete button is clicked. 

I'll redo it with your code as that looks better I think